### DISCUSSION TITLE
Mounting persistent volumes for code-server using Docker

### DISCUSSION THREAD

**Discussion by RoadToML** [coder/code-server#4869] [Q&A]
Hey, we have coder-server running docker and I was wondering how to mount a folder from my NAS so that it loads in every workspace that is built. 

thank you! :)

**Comment by jsjoeio** [comment]
Hmm...could you copy it to [this directory](https://github.com/coder/code-server/blob/main/ci/release-image/Dockerfile#L47)?

Then when the workspace loads, the folder would be there, right?

**Reply by RoadToML** [reply]
Copying seems impractical as the NAS hosts a dataset that spans 10s of TBs so the best solution would be is to mount the NAS

**Reply by jsjoeio** [reply]
Ah okay I have no idea then 😅 Let's ask someone who knows Docker better than me. @jawnsy or @bpmct maybe?

**Reply by jawnsy** [reply]
You can use a Docker volume to do this, but I can't really help as I don't have much personal experience using it. When you run code-server, run with a volume (-v flag), and pre-define a volume using the instructions here: https://docs.docker.com/storage/volumes/#share-data-among-machines - you need a volume driver that can mount your NAS partition, maybe NFS or similar.

As an alternative, you can also mount a volume to your host system and then bind-mount it into the container (e.g. mount your NFS at /mnt/nfs) and then run with `docker run -v /mnt/nfs:/mnt/nfs:rw code-server`

**Reply by RoadToML** [reply]
Will give this a go and let you know. Cheers!
